src/txn/txn.c

```diff
@@ -1005,6 +1005,8 @@ __wt_txn_global_shutdown(WT_SESSION_IMPL *session)
 		if (txn_global->oldest_id == txn_global->current &&
 		    txn_global->metadata_pinned == txn_global->current)
 			break;
+
+		WT_STAT_CONN_INCR(session, txn_release_blocked);
```

Comment:
Used the counter only in performance critical paths, else directly updating the statistics.